### PR TITLE
Closes #3526: refactor argsortMsg to remove registerND annotation

### LINKED ISSUE #3526: refactor argsortMsg to remove registerND annotation
Refactor argsortMsg to remove @registerND annotation, so that annotation can eventually be deprecated.

### PR DESCRIPTION
**Pull request by ajpotts**
This PR refactors `argsortMsg` to use `@arkouda.instantiateAndRegister`.

### CODE REVIEW

File: src/ArgSortMsg.chpl [hunk 1]
```diff
@@ -404,95 +424,47 @@ module ArgSortMsg
...
+    proc argsort(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+      where array_dtype == BigInteger.bigint
+    {
+        return MsgTuple.error("argsort does not support the bigint dtype");
+    }
+
+    proc argsort(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+      where (array_dtype == uint(8)) 
+    {
+        return MsgTuple.error("argsort does not support the unit(8) dtype");
```
Comment (jeremiah-corrado) [inline src/ArgSortMsg.chpl:451]: both of these overloads could be combined with:

```chapel
return MsgTuple.error("argsort does not support the %s dtype".format(array_dtype:string));
```

**Review by jeremiah-corrado — APPROVED** [review]
Looks good! Just one minor suggestion:

File: src/ArgSortMsg.chpl (hunk 1 above)
Comment (ajpotts) [reply src/ArgSortMsg.chpl]: Great suggestion!  I made the change.  Thanks.

**Review by stress-tess — APPROVED** [review]
looks great!! thanks @ajpotts!

**Review by jaketrookman — APPROVED** [review]
Looks great

**Pull request merged**